src/view.ts

```diff
@@ -308,5 +311,6 @@ export default class CalendarView extends ItemView {
     await leaf.openFile(existingFile, { mode });
 
     activeFile.setFile(existingFile);
+    workspace.setActiveLeaf(leaf, true, true)
```

Comment:
There is a slightly more succinct way to write this (granted it's missing from the API spec currently).

`leaf.openFile()` takes a second parameter for passing in the ephemeral state. So it could be:

```suggestion
    await leaf.openFile(existingFile, { active: true, mode });
```

Reply:
thank you, didn't know that
fixed → https://github.com/liamcain/obsidian-calendar-plugin/pull/178/commits/d6d5475b10aa8cc6de9203cb571d811d9a191ac1